### PR TITLE
Updated Paymentwall lib

### PR DESCRIPTION
**Pull request by dzungtran**

### CODE REVIEW

File: upload/library/XenForo/UserUpgradeProcessor/Paymentwall.php [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+        $this->_upgradeModel->logProcessorCallback(
+            $upgradeRecordId, $processor, $transactionId, $type, $message, $details
+        );
+
+        if (!$this->alreadyprocessed) {
+            switch ($this->_filtered['type']) {
+                case self::PAYMENT_CHARGE_BACK:
+                    $type = 'cancelled';
+                    break;
+                case '1':
```
Comment (sevastyanovio) [inline upload/library/XenForo/UserUpgradeProcessor/Paymentwall.php]: use const

File: upload/library/XenForo/UserUpgradeProcessor/Paymentwall.php [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+                    $type = bdPaygate_Processor_Abstract::PAYMENT_STATUS_ACCEPTED;
+                    break;
+                case '0':
```
Comment (sevastyanovio) [inline upload/library/XenForo/UserUpgradeProcessor/Paymentwall.php]: use const

File: upload/paymentwall.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+try {
+    if (!$processor->validateRequest($logMessage)) {
+        $logType = 'error';
+
+        $response->setHttpResponseCode(500);
+    } else if (!$processor->validatePreConditions($logMessage)) {
+        $logType = 'error';
+    } else {
+        list($logType, $logMessage) = $processor->processTransaction();
+    }
```
Comment (tridungpham) [inline upload/paymentwall.php]: Honestly I don't think the two methods validateRequest() and validatePreConditions() are good practices. Your function should only check for errors and returns boolean, but not modify logMessage. Actually we don't need to pass log message to those methods.

the code above can be like this :

``` php
try {
    if (!($processor->validateRequest() && $processor->validatePreConditions())) {
        throw new Exception($processor->getLogMessage(), 500);
    }

    list($logType, $logMessage) = $processor->processTransaction();
} catch (Exception $e) {
    $response->setHttpResponseCode($e->getCode());
    // ... 
}

$processor->log($logType, $logMessage, $logExtra);
```

in class XenForo_UserUpgradeProcessor_Paymentwall you shoud define a method to retrieve log message

Comment (sevastyanovio) [inline upload/paymentwall.php]: :+1:

Comment (dzungtran) [inline upload/paymentwall.php]: Thanks :)

File: src/upload/library/Paymentwall/Processor.php [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+            array(
+                'email' => XenForo_Visitor::getInstance()->get('email'),
+                'integration_module' => 'xenforo',
+                'test_mode' => $this->options->get('Paymentwall_testMode'),
+                'success_url' => $this->_generateReturnUrl($extraData),
+                'custom' => $custom,
+            ));
+
+        $callToAction = new XenForo_Phrase('Paymentwall_call_to_action');
+        //$_xfToken = XenForo_Visitor::getInstance()->get('csrf_token_page');
```
Comment (sevastyanovio) [inline src/upload/library/Paymentwall/Processor.php]: Remove

**Pull request closed without merging**